packages/financial-templates-lib/src/price-feed/USPACPriceFeed.ts

```diff
@@ -0,0 +1,214 @@
+   */
+  constructor(
+    private readonly logger: Logger,
+    private readonly web3: Web3,
+    private readonly symbols: string[],
+    private readonly correctionFactor: string,
+    private readonly marketstackApiKey: string,
+    private readonly useHttps: boolean,
+    private readonly lookback: number,
+    private readonly networker: NetworkerInterface,
+    private readonly getTime: () => Promise<number>,
+    private readonly priceFeedDecimals = 4 /* Prices in Marketstack are 2 decimals, but give some extra precision */,
+    private readonly minTimeBetweenUpdates = 43200 // 12 hours is a reasonable default since this pricefeed returns daily granularity at best.
+  ) {
+    super();
+
+    assert(symbols != null, "symbols must be provided");
+    assert(marketstackApiKey != null, "marketstackApiKey must be provided");
+    assert(correctionFactor != null, "correctionFactor must be provided");
+
+    this.uuid = `USPACPriceFeed-${symbols.join(",")}`;
+  }
+
+  public getCurrentPrice(): BN | null {
+    return this.currentPrice;
+  }
+
+  public async getHistoricalPrice(time: number, ancillaryData: string, verbose = false): Promise<BN> {
+    return this.fetchPrice(time, verbose);
+  }
+
+  public getLastUpdateTime(): number | null {
+    return this.lastUpdateTime;
+  }
+
+  public getLookback(): number {
+    return this.lookback;
+  }
+
+  public getPriceFeedDecimals(): number {
+    return this.priceFeedDecimals;
+  }
+
+  public async update(): Promise<void> {
+    const currentTime = await this.getTime();
+
+    // Return early if the last call was too recent.
+    if (this.lastUpdateTime !== null && this.lastUpdateTime + this.minTimeBetweenUpdates > currentTime) {
+      this.logger.debug({
+        at: "USPACPriceFeed",
+        message: "Update skipped because the last one was too recent",
+        currentTime: currentTime,
+        lastUpdateTimestamp: this.lastUpdateTime,
+        timeRemainingUntilUpdate: this.lastUpdateTime + this.minTimeBetweenUpdates - currentTime,
+      });
+      return;
+    }
+
+    this.logger.debug({
+      at: "USPACPriceFeed",
+      message: "Updating USPACPriceFeed",
+      currentTime: currentTime,
+      lastUpdateTimestamp: this.lastUpdateTime,
+    });
+
+    this.currentPrice = await this.fetchPrice(currentTime, false);
+    this.lastUpdateTime = currentTime;
+  }
+
+  private async fetchPrice(time: number, verbose: boolean): Promise<BN> {
+    // Find the closest day that completed before the beginning of the lookback window, and use
+    // it as the start date.
+    // Marketstack API does not return data for non-working days. We must give
+    // lookback that is big enough to cover at least one working day. For
+    // example, NYSE exchange can have maximum three consecutive non-working
+    // day. We make some reserve and set 10 days as minimal lookback.
+    const MINIMAL_LOOKBACK = 3600 * 24 * 10;
+    const startLookbackWindow = time - Math.max(this.lookback, MINIMAL_LOOKBACK);
```

Comment:
Normally in other price feed implementations `lookback` refers to time from when liquidable position is being evaluated to current time. In order not to cause confusion by the consuming applications it should be renamed to different parameter. As an example, [TraderMadePriceFeed.ts](https://github.com/UMAprotocol/protocol/blob/master/packages/financial-templates-lib/src/price-feed/TraderMadePriceFeed.ts) has `minuteLookback` and `hourlyLookback` parameters while [CryptoWatchPriceFeed.ts](https://github.com/UMAprotocol/protocol/blob/master/packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.ts) has `historicalTimestampBuffer` parameter to allow using earlier timestamp if the price is not available at request timestamp.

Reply:
I pushed new commit that removes this code, is your comment still actual?